src/app/webdav/cloud-file-service.service.spec.ts

```diff
@@ -1,23 +1,72 @@
 import { TestBed } from '@angular/core/testing';
-
 import { CloudFileService } from './cloud-file-service.service';
 import { SessionService } from 'app/session/session.service';
+import { User } from 'app/user/user';
+import { AppConfig } from 'app/app-config/app-config';
+import webdav from 'webdav';
 
 describe('CloudFileService', () => {
-  beforeEach(() => TestBed.configureTestingModule({}));
+  let cloudFileService: CloudFileService;
+  let sessionService: jasmine.SpyObj<SessionService>;
+  let sessionSpy;
+  let clientSpy;
 
   beforeEach(() => {
-    TestBed.configureTestingModule({
-      providers: [
-        SessionService
-      ],
-    })
-    .compileComponents();
-  });
-
-  // TODO: write tests!
-  // it('should be created', () => {
-  //   const service: CloudFileService = TestBed.get(CloudFileService);
-  //   expect(service).toBeTruthy();
-  // });
+    AppConfig.settings = {
+      site_name: '',
+      database: {name: 'unit-tests', remote_url: '', timeout: 60000, outdated_threshold_days: 0, useTemporaryDatabase: true},
+      webdav: {remote_url: 'test-url'}
+    };
+
+    sessionSpy = jasmine.createSpyObj('SessionService', ['getCurrentUser']);
+    clientSpy = jasmine.createSpyObj('client', ['getDirectoryContents', 'createDirectory', 'getFileContents', 'putFileContents']);
+
+    TestBed.configureTestingModule({ providers: [
+      CloudFileService,
+      { provide: SessionService, useValue: sessionSpy}]
+    });
+
+    cloudFileService = TestBed.get(CloudFileService);
+    cloudFileService['client'] = clientSpy;
+    sessionService = TestBed.get(SessionService);
+  });
+
+  it('.connect() should call sessionService.getCurrentUser()', () => {
+    cloudFileService.connect('user', 'pass');
+    expect(sessionService.getCurrentUser).toHaveBeenCalled();
```

Comment:
is the call to `getCurrentUser` really the relevant thing to test? Isn't the call to webdav or the creation of the client object what really matters from an outside perspective?